backends/gstreamer/src/audio_decoder.rs

```diff
@@ -67,12 +67,30 @@ impl AudioDecoder for GStreamerAudioDecoder {
 
         let pipeline_ = pipeline.clone();
         let callbacks_ = callbacks.clone();
+        // Initial pipeline looks like
+        //
+        // appsrc ! decodebin2! ...
+        //
+        // We plug in the second part of the pipeline, including the deinterleave element,
+        // once the media starts being decoded.
         decodebin.connect_pad_added(move |_, src_pad| {
-            // We only want a sink per audio stream.
+            // Ignore any additional source pads just in case.
             if src_pad.is_linked() {
```

Comment:
This is never going to be true. What you want to check here is if linking to the next sinkpad fails because the pad is already linked